subprojects/language-java/src/main/java/org/gradle/api/internal/tasks/compile/incremental/deps/ClassSetAnalysis.java

```diff
@@ -103,8 +103,8 @@ public DependentsSet getRelevantDependents(String className, IntSet constants) {
         if (!constants.isEmpty()) {
             return DependentsSet.dependencyToAll();
         }
-        Set<String> classesDependingOnAllOthers = annotationProcessingData.getGeneratedTypesDependingOnAllOthers();
-        Set<GeneratedResource> resourcesDependingOnAllOthers = annotationProcessingData.getGeneratedResourcesDependingOnAllOthers();
+        Set<String> classesDependingOnAllOthers = Sets.union(annotationProcessingData.getAggregatedTypes(), annotationProcessingData.getGeneratedTypesByOrigin().keySet()).contains(className) ? annotationProcessingData.getGeneratedTypesDependingOnAllOthers() : Collections.emptySet();
```

Comment:
This seems to be done in a loop. Should we cache the union in `annotationProcessingData`?

Reply:
I don't think it's a big deal here since it's a live set, not an actual computation of union.

Reply:
Updated in any case.